### PR TITLE
Add a short description as well as build instructions in repo's README.md

### LINKED ISSUE #165: Document build/install process
We should have documentation on building/installing the project (using `autotools`).

### PR DESCRIPTION
**Pull request by fvalasiad**
Closes #165

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,6 +5,43 @@ to fully record the interactions
...
+To build it you are going to need the following tools:
+* A C compiler
+* make
+* autoconf
+* automake
+
+As well as the following libraries:
+* libcrypto
+
+### Build Instructions
```
Comment (zvr) [inline README.md:37]: This is "build from the git repo".

A similar one would be "build from the released tarball" (without `autoreconf`).

File: README.md [hunk 2]
```diff
@@ -5,6 +5,43 @@ to fully record the interactions
 between assets (files and tools)
 when a software artifact is being built (compiled).
 
+## Synopsis
```
Comment (zvr) [inline README.md:8]: Call it "Syntax" or something.

The "Synopsis" is specialized term, and will only be used on the man page.

**Review by zvr — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (fvalasiad) [reply README.md:8]: @zvr What about "Usage"?

Comment (zvr) [reply README.md]: Sure, that's fine.

**Pull request merged**